[CS2113-T10-3] MangaTantou

xenthm opened the pull request:
MangaTantou serves to manage the tasks under which editors in a manga company have to handle. This includes having to chase authors for deadlines, track sales revenue, manage authors’ salaries etc.

thoriritu commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,359 @@
+import static constants.Regex.PRICE_OPTION_REGEX;
+import static constants.Regex.QUANTITY_OPTION_REGEX;
+import static constants.Regex.SPACE_REGEX;
+
```
Consider using enums instead of static constants like CATALOG_COMMAND to improve maintainability.

thoriritu commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,359 @@
+     *                  which should specify an Add operation for either Manga or Author
+     * @return a Command object representing the Add operation for Manga or Author
+     * @throws TantouException if the user input is invalid for both Manga and Author Add commands
+     */
```
consider extracting code (within each block) to individual methods if possible

thoriritu commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,359 @@
+            return new AddSalesCommand(salesArguments);
+        }
+        throw new TantouException("Invalid sales command provided!"
+                + " You need to provide the author, manga, quantity sold, and unit price.");
```
Consider splitting the class as it has many functionality and can be separated into 2 different ones.

thoriritu commented on src/main/java/storage/MangaDeserializer.java:
```diff
@@ -0,0 +1,56 @@
+            throw new JsonParseException("corrupt Manga object");
+        }
+        JsonObject mangaJsonObject = json.getAsJsonObject();
+
```
Instead of generic messages like "corrupt manga name", provide more informative ones that specify which field was invalid.

xenthm replied:
Thanks for the feedback! The part you are referencing checks if the `Manga` entry in the data file is a `JsonObject` (whichs means it shouldn't be formatted as an array or anything else. I provided more informative messages if the `mangaName` or `deadline` contains issues in the following lines of code.

thoriritu commented on src/main/java/commands/DeleteMangaCommand.java:
```diff
@@ -0,0 +1,86 @@
+                saveFile(authorList);
+                return;
+            }
+            assert !existingAuthor.hasManga(deletingManga): "No manga found";
```
The assertion check for finding the existence of authorName after already validating it before could be removed since they are unnecessary

chewycj commented on docs/uml/images/mangasales_class.png:
The class diagram is really detailed and well-drawn! 🚀 However, there were some non-standard notation for the diagram `mangasales_class.png` to note, according to prof, I think can refer to how the team did the command.png for the standard format too! 👍 

Class name:
![image](https://github.com/user-attachments/assets/d9d30ffc-10b5-4808-8583-bb50abd98db2)
Perhaps for normal classes, I think we don't need to coloured green C!
such as from the addressbook example:
![image](https://github.com/user-attachments/assets/361929bc-dc02-4ed4-a7ea-08eeff45f196)

Visibility:
![image](https://github.com/user-attachments/assets/e10922b4-c0ed-4f2c-8f55-e64ce9d23494)
The symbols for the visibility could be changed to '+' and '-' or '#' accordingly too!

xenthm replied:
Yes! This will be fixed once we include our standardised config file to the puml files.

chewycj commented on docs/uml/images/AddAuthorSequence.png:
All of the team's sequence diagrams are well drawn and neat! 💯 One thing to note that prof has pointed out is that the boxes of the entities are rounded, as it should be a sharp corner instead!

For example, these are rounded:
![image](https://github.com/user-attachments/assets/d6295c52-c9ea-4aca-81ab-1f635a15de32)

averageandyyy replied:
Noted on this, thanks for catching it!

chewycj commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,269 @@
+> **_NOTE:_** There is circular reference (bidirectional navigability) between `Author` and `Manga` through `MangaList`. 
+
+### Commands 
+![Command Inheritance](/docs/uml/puml/Command/Command.png)
```
Perhaps the Command.png can be placed in the images folder! As currently it could not load in the DG 😿 

![image](https://github.com/user-attachments/assets/63e3f3d7-23cd-4959-b36b-0c76ccf2cbf6)

averageandyyy replied:
Ah, careless mistake! Thanks for helping us point it out early!

chewycj commented on docs/uml/puml/AuthorListClass/AuthorListClass.png:
Perhaps the UML note can be connected by a dotted line instead!

![image](https://github.com/user-attachments/assets/179fd9b7-1a4b-485a-a1d9-227733bcc970)

xenthm replied:
Thank you, will look into that!

chewycj reviewed (commented):
Overall, the diagrams are really clear and detailed!! 💯 Just some minor notations to change 😄 We found this from a past year forum that might be useful for formatting the diagrams: https://github.com/nus-cs2113-AY2324S2/forum/issues/35

jemehgoh commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,286 @@
+
+#### Guidelines for Future Developers
+When adding new command classes, developers must follow the same method of implementation by 
+inheriting from the abstract `Command` class. Ensure that each new command class includes 
+an implementation of the `execute` method and appropriately interacts with the `Ui` class 
```
I think you should not repeat yourself here regarding implementing the "execute" method?

averageandyyy replied:
Thanks for the comment, I agree that the content is similar in nature regarding the `execute` method but in the Guidelines, I had phrased it and targetted my language to be instructional towards the developers while in Structure, it is more of a generic description. Nonetheless, I hear you and will look into making the text more targetted with reduced repetition. Thanks again!

jemehgoh commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,286 @@
+For the AddSalesCommand to be successful, the manga that the sales data is associated with must exist. If the `sales`
+command is successful, the `Sales` data is then saved via Storage.
+
+![mangasales_class.png](uml/images/mangasales_class.png)
```
I think this class diagram is not relevant to this section and shouldn't be placed here?

jemehgoh commented on docs/uml/puml/AuthorListClass/AuthorListClass.puml:
```diff
@@ -0,0 +1,46 @@
+    + getAuthor(authorName: String): Author
+}
+
+class ArrayList<T>
+ArrayList <|-- AuthorList
+note on link : ArrayList<Author>
+ArrayList <|-- MangaList
+note on link : ArrayList<Manga>
```
![Screenshot 2024-10-30 132426](https://github.com/user-attachments/assets/991bbc44-a203-415b-8f97-c46f2f15afc5)
In this diagram, MangaList and AuthorList are indicated as inheriting from ArrayList, but with notes stating "ArrayList<Manga>" and "ArrayList<Author>". I think that, unless these two classes inherit from ArrayList<T> (in which I think you should remove the notes), I believe that there is no inheritance to indicate?

xenthm replied:
Sorry, I'm not sure I understand you. They both inherit from `ArrayList`, but `AuthorList` extends `ArrayList<Author>`, while `MangaList` extends `ArrayList<Manga>`. I still do believe the notes are necessary for show that distinction.

xenthm replied:
Will be keeping this representation based on: 

> Your notation seems OK (* good!). Not sure what the comment means specifically.
> 
> From UML specification: 
> ![image](https://github.com/user-attachments/assets/29424410-ceb6-45c7-bc60-db6798217647)
> 
> *we don't cover this in CS2113; so you went beyond to learn the notation for generics inheritance. 
> specifying the type in the UML note seems OK to me as a way to explain to the reader what it means. 

_Originally posted by Prof in https://github.com/nus-cs2113-AY2425S1/forum/issues/41#issuecomment-2453045543_

jemehgoh commented on docs/uml/puml/ViewMangaSequence/ViewMangaSequence.puml:
```diff
@@ -0,0 +1,33 @@
+        cmd -> ui : print list as table with the required columns
+            ref over ui, col : Print all required list and format rows
+        return
+return done printing
```
![image](https://github.com/user-attachments/assets/8d53f0f6-59a9-4b50-aaba-3ea78f340d82)

I think you should remove the "done printing" label, as the dotted arrow is sufficient to indicate that control has gone back to Tantou?

jemehgoh commented on docs/uml/puml/ViewMangaSequence/refGetColumnsToPrintSequence.puml:
```diff
@@ -0,0 +1,44 @@
+group sd [Add required columns to list]
+    ml -> row ** : add new PrintColumn
+    row ++
+    return added to columns
```
<img width="396" alt="image" src="https://github.com/user-attachments/assets/eec06324-ca68-4ba2-970e-72718da0d04d">

I think you don't have to mention "added to columns" every time you return control, since you already mention "add new [column]" for each corresponding operation?

xenthm replied:
You're right. Perhaps I can make it clearer by appending "to columns" to each top arrow.

jemehgoh reviewed (commented):
Good job on your guide! The explanations are mostly relevant and the diagrams are mostly appropriate. I just have several nits, as mentioned in my other comments.

lowjunchen commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,269 @@
+
+#### Storage Structure
+![StorageClass.png](uml/puml/StorageClass/StorageClass.png)
+The above UML class diagram outlines the structure of the classes related to saving data. 
```
This line is not well formattted. I think it is meant to start on the next line

lowjunchen commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,269 @@
+#### Interaction
+The following diagram illustrates the interactions that take place when the
+user provides `"catalog -a Kubo Tite"` as an input.
+![add author sequence diagram](/docs/uml/images/AddAuthorSequence.png)
```
This is a broken link which is not visible.

lowjunchen commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,269 @@
+For the AddSalesCommand to be successful, the manga that the sales data is associated with must exist. If the `sales`
+command is successful, the `Sales` data is then saved via Storage.
+
+![mangasales_class.png](uml/images/mangasales_class.png)
```
May I clarify what is the meaning of the circle and square icons? If I remember correctly it should be + or - or omitted

xenthm replied:
It is the default way for PlantUML to display the access modifier. But you are correct, we should be following the textbook for this course.

lowjunchen commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,269 @@
+
+The following sequence diagram illustrates the interactions that occur when the parser creates a new `AddSalesCommand`.
+
+![addsalesdata.png](uml/images/addsalesdata.png)
```
Is the sequence diagram format correct? Should it not have the instance at the bottom?
![image](https://github.com/user-attachments/assets/64ed3324-36cf-4542-90f8-1125971c1a50)

xenthm replied:
You are right, it should not be there. Thank you.

averageandyyy committed:
Rename MangaTantou to Tantou

averageandyyy committed:
Update Main to use Tantou

iaso1774 committed:
Merge branch 'master' into schedule_updated_parser_fixed

Commit:
Fixed import issues caused by branch merge

Commit:
Removed unused import

iaso1774 committed:
Merge pull request #99 from iaso1774/schedule_updated_parser_fixed

Updated schedule to work with the new parser

Commit:
Included config file based on suggestion by @xenthm

Commit:
Included config file based on suggestion by @xenthm

iaso1774 committed:
Merge branch 'master' into DG

Commit:
Updated schedule.png

Commit:
Merge remote-tracking branch 'origin/DG' into DG

iaso1774 committed:
Merge pull request #101 from iaso1774/DG

DG - Schedule

averageandyyy committed:
Update Deadline tests to check for the right exception being thrown

averageandyyy committed:
Remove unused import

averageandyyy committed:
Update Tantou references to MangaTantou

averageandyyy committed:
Update parsing regex to stop at all valid flags, flag of interest included

xenthm committed:
Merge pull request #114 from averageandyyy/branch-StressTest

Branch stress test

xenthm committed:
Update UML diagrams to match new Sale class

xenthm committed:
Update DG

- Wrap long lines
- Fill in Product Scope section
- Fill in User Stories section
- Fill in Non-Functional Requirements section
- Add instructional placeholder text for Manual Testing section

xenthm committed:
Add table of contents to DG

xenthm committed:
Add table of contents to DG

xenthm committed:
Update DG with new data file format

xenthm committed:
Add field extractor regex explanation

xenthm committed:
Update capitalization of manual testing section

xenthm committed:
Update AboutUs.md

- Added more information about us and the course
- Fixed the formatting of the table

sarahchow03 committed:
Update AddSalesCommand sequence diagram

sarahchow03 committed:
Refactor .img and .puml names

sarahchow03 committed:
Add generalizable sequence diagram for command processing

xenthm committed:
Update spelling for regex to RegEx

# Conflicts:
#	docs/DeveloperGuide.md

xenthm committed:
Add Donovan's (xenthm) PPP

xenthm committed:
Update link to PPP in AboutUs.md

xenthm committed:
Cut some words from PPP

xenthm committed:
Change log message output to a file

Log file is located in logs/tantou.log. Made a custom log message
format. Also added the logs directory to .gitignore.

xenthm committed:
Perform checks if logging is enabled in Tantou

xenthm committed:
Remove log file creation messages

It messes with the I/O tests, and the team decided that the log file can
be created without alerting the user.

xenthm committed:
Remove unused imports in Main

averageandyyy committed:
Merge pull request #115 from xenthm/branch-dg-updates

DG updates

averageandyyy committed:
Merge pull request #121 from xenthm/branch-log-file

Change log message output to a file

averageandyyy committed:
Update AddAuthorCommand sequence diagram

averageandyyy committed:
Update AddMangaCommand sequence diagram

averageandyyy committed:
Update DeleteAuthorCommand sequence diagram

averageandyyy committed:
Update code to match sequence diagrams
- Introduce methods to make use of Ui to interact with user instead of directly souting
- Replace TantouExceptions with their more specific counterparts

averageandyyy committed:
Fix typos in sequence diagram

averageandyyy committed:
Update tests

sarahchow03 committed:
Edit portfolio link on AboutUs.md

sarahchow03 committed:
Fix cosmetics issues on UserGuide.md

sarahchow03 committed:
Add profile template

sarahchow03 committed:
Update based on PR comments

averageandyyy committed:
Add PPP

averageandyyy committed:
Add details regarding testing

averageandyyy committed:
Add placeholder for DG testing contribution

sarahchow03 committed:
Fix order of getMangaName in sequence diagram

sarahchow03 committed:
Fix label on StorageHelper

sarahchow03 committed:
Fix .png imports

sarahchow03 committed:
Fix MangaSalesClass to excluded totalRevenue

averageandyyy committed:
Update AddAuthorCommand sequence diagram

averageandyyy committed:
Update AddAuthorCommand sequence diagram

averageandyyy committed:
Update DeleteAuthorCommand sequence diagram

averageandyyy committed:
Update DeleteMangaCommand sequence diagram

averageandyyy committed:
Update DeleteManga sequence diagram

xenthm committed:
Merge pull request #123 from averageandyyy/branch-UpdatePPPAndDG

Branch update ppp and dg

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master' into branch-developerGuideFix

# Conflicts:
#	docs/DeveloperGuide.md

sarahchow03 committed:
Update AddSalesCommand after merge conflict fix

sarahchow03 committed:
Correct 'Int' to 'Integer'

xenthm committed:
Merge pull request #117 from sarahchow03/branch-developerGuideFix

Branch developer guide fix

Commit:
Merge branch 'master' of https://github.com/iaso1774/tp

# Conflicts:
#	docs/uml/puml/schedule.puml

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Rewrote schedule SD to better align with existing SD's

Commit:
Removed commented out code

sarahchow03 committed:
Remove unnecessary throws in ArgumentFinder

sarahchow03 committed:
Fix checkStyle errors

sarahchow03 committed:
Update JavaDoc comments for ArgumentFinders

Commit:
Made changes suggested in Telegram conversation

Commit:
Added image file

sarahchow03 committed:
Remove unused exceptions

sarahchow03 committed:
Update JavaDoc comments for AddSalesCommand

sarahchow03 committed:
Remove unused code and imports

sarahchow03 committed:
Update README.md

xenthm committed:
Merge pull request #127 from sarahchow03/branch-fixException

Branch fix exception

xenthm committed:
Merge pull request #126 from iaso1774/schedule_sd_update

Schedule SD update

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master'

xenthm committed:
Remove commented-out lines from DG

xenthm committed:
Add newline to UG to try and fix table format

xenthm committed:
Update UG

- Replace all instances of "%2F" to "/"
- Fix paths of some UML diagrams

xenthm committed:
Add FAQ for UG

xenthm committed:
Add section for schedule in UG

xenthm committed:
Update DG table of contents

xenthm committed:
Update title capitalization for DG

sarahchow03 committed:
Merge pull request #129 from xenthm/branch-fix-docs

Branch fix docs

xenthm committed:
Update link to Command Inheritance file

xenthm committed:
Update references to AddDeadlineCommand in DG

xenthm committed:
Merge pull request #130 from xenthm/branch-fix-docs

Update link to Command Inheritance file

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master'

sarahchow03 committed:
Add note on case-sensitivity of names

Commit:
Fixed issues brought up with Schedule in the PED.

sarahchow03 committed:
Add item to FAQ

sarahchow03 committed:
Add note on duplicate flag operations

sarahchow03 committed:
Improve clarity of AddSalesCommand

sarahchow03 committed:
Add 'bye' to Command Summmary

sarahchow03 committed:
Address 'edit' function in FAQ

sarahchow03 committed:
Remove guide text

sarahchow03 committed:
Improve clarity on case-sensitivity

sarahchow03 committed:
Fix grammatical errors

iaso1774 committed:
Merge pull request #171 from sarahchow03/branch-uGfixes

User guide fixes for PE-D

Commit:
Ian's PPP

sarahchow03 committed:
Specify number format exceptions

sarahchow03 committed:
Update NumberLessThanZeroException

xenthm committed:
Add BASE_LOCATION relative to jar file in Tantou

Only set if MangaTantou is run via a jar file. If not, the working
directory is used for data and log files.

xenthm committed:
Update Storage.java

- Use Tantou.BASE_LOCATION
- Differentiate log messages for initialization and change of data file
  path
- Remove "singleton initialized" log message

xenthm committed:
Update UG FAQ

To reflect new instructions to transfer data and logs.

xenthm committed:
Remove redundant sout message in ViewMangasCommand

xenthm committed:
Update Tantou.java

- Add author tags
- Remove sout message when jar file detected
- Add more details to sout message if URISyntaxException thrown

sarahchow03 committed:
Fix cosmetic errors in DeveloperGuide.md

sarahchow03 committed:
Update CommandSequence with corrections

sarahchow03 committed:
Clarify usage of AddSalesCommand

xenthm committed:
Update Author and Manga Deserializers

Inform user to check for redundant commas in the thrown exception
message.

xenthm committed:
Update AuthorList and MangaList Deserializers

- Change enhanced for loops to standard one (with int i = 0)
- Add index information in sout message to user when data restoration
  fails for a certain entry
- Manually use the AuthorDeserializer for Author Json elements in
  AuthorListDeserializer

xenthm committed:
Remove redundant registerTypeAdapter call in Storage

Because the AuthorDeserializer is manually set in
AuthorListDeserializer, this can be removed.

xenthm committed:
Extract maximum values for sales data into Options.java

xenthm committed:
Refactor exceptions to use extracted constants

averageandyyy committed:
Merge pull request #174 from xenthm/fix-repeat-exception-message

Remove redundant sout message in ViewMangasCommand

xenthm committed:
Add MANIFEST.MF to gitignore

averageandyyy committed:
Merge pull request #173 from xenthm/fix-file-location

Make data and log files save to path relative to jar file

Commit:
Updated note based on feedback from @sarahchow03

Commit:
Updated PPP based on comment from @averageandyyy

iaso1774 committed:
Merge pull request #172 from iaso1774/Ian's-PPP

Ian's PPP

averageandyyy committed:
Refactor printing of successful addition of deadline into Ui method

averageandyyy committed:
Refactor printing of successful addition of sales into Ui method

averageandyyy committed:
refactor printing of author list size message into ui method

averageandyyy committed:
refactor printing of mangalist message into ui method

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master' into branch-commandSequenceFix

averageandyyy committed:
Refactor printing messages in Tantou

sarahchow03 committed:
Add CommandSequence to DeveloperGuide

sarahchow03 committed:
Update AddSalesCommandTest

averageandyyy committed:
Refactor sout methods in Storage

averageandyyy committed:
Refactor sout methods in SaleDeserializer

averageandyyy committed:
Refactor sout methods in MangaListDeserializer

averageandyyy committed:
refactor sout method in AuthorListDeserializer

averageandyyy committed:
Merge pull request #175 from sarahchow03/branch-commandSequenceFix

Fix CommandSequence diagram

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master'

averageandyyy committed:
Complete half of command sequence diagram

averageandyyy committed:
Update command one

averageandyyy committed:
Create command 2

averageandyyy committed:
Create command three

averageandyyy committed:
Update Parser overall structure textual content

averageandyyy committed:
Scale up Parser image test

averageandyyy committed:
Scale image test

averageandyyy committed:
Break up parser image into two

sarahchow03 committed:
Add contributions to PPP

averageandyyy committed:
Update aboutus to link to correct PPP

averageandyyy committed:
Update TOC

averageandyyy committed:
Update PPP

sarahchow03 committed:
Correct directory name for PPP

averageandyyy committed:
Merge pull request #179 from sarahchow03/branch-PPP

Sarah's PPP

xenthm committed:
Add overload for hasManga in Author.java

Checks if the provided mangaName String is found in the mangaList.

xenthm committed:
Extract validity checks from Command subclasses

Created the CommandValidity utility class.

xenthm committed:
Make COMMAND_LOGGER package-private

So that the CommandValidity class can access it from within the comands
package.

averageandyyy committed:
Merge github.com:AY2425S1-CS2113-T10-3/tp into branch-FixDGImages

averageandyyy committed:
Update aboutus

xenthm committed:
Extract max deadline string length

From PrintFormat to Options.

xenthm committed:
Add deadline too long exception and validity checks

- Created custom DeadlineTooLongException
- Added null/empty and length checks to CommandValidity

xenthm committed:
Remove * imports from CommandValidity

sarahchow03 committed:
Merge pull request #178 from averageandyyy/branch-FixDGImages

Branch fix dg images

xenthm committed:
Refactor AddAuthorCommand

xenthm committed:
Combine methods in CommandValidity

- Combine null/empty check and length check for author name, manga name,
  and deadline into one method
- Make combined methods private in CommandValidity
- Refactor usage in AddAuthorCommand

xenthm committed:
Refactor AddDeadlineCommand

xenthm committed:
Rename CommandValidity method names

To better fit what they are doing.

xenthm committed:
Refactor AddMangaCommand

sarahchow03 committed:
Specify numbers to be 'non-negative'

xenthm committed:
Refactor AddSalesCommand

xenthm committed:
Refactor DeleteAuthorCommand

xenthm committed:
Refactor DeleteMangaCommand

xenthm committed:
Refactor ViewAuthorsCommand

xenthm committed:
Refactor ViewMangasCommand

xenthm committed:
Rename check to ensure in CommandValidity

To better fit what the methods are doing

xenthm committed:
Fix checkstyle

xenthm committed:
Remove redundant logs in Commands and change format

Format now enclosed author and manga names with double quotation marks
for easier distinction between name and log message.

xenthm committed:
Make COMMAND_LOGGER in Command public

xenthm committed:
Use different log formatter for Tantou.Commands

This allows the log message to contain the caller's (i.e. the command's)
name, instead of the CommandValidity utility class's name.

xenthm committed:
Add author tags in Tantou

xenthm committed:
Add author name check to AuthorListDeserializer

xenthm committed:
Merge pull request #176 from sarahchow03/branch-fixCommandException

Fix command exception messages

xenthm committed:
Merge remote-tracking branch 'upstream/master' into branch-validity-checks

xenthm committed:
Update messages shown to users

To better fit existing message formats.

xenthm committed:
Extract unit price and quantity maximums

Into Options and PrintFormat.

xenthm committed:
Extract sales data number validity checks

Into CommandValidity.

xenthm committed:
Use constants package to print message in exceptions

- PriceTooLargeException: MAX_UNIT_PRICE_VALUE
- QuantityTooLargeException: MAX_QUANTITY_VALUE

xenthm committed:
Add part about integer overflow in exception

QuantityWrongFormatException

xenthm committed:
Fix checkstyle

xenthm committed:
Fix JUnit tests for deserializers

xenthm committed:
Remove unused constants from AddSalesCommand

xenthm committed:
Merge branch 'fix-manual-data' into branch-validity-checks

xenthm committed:
Split ensureValidSalesDataNumbers method into two

For better SRP.

xenthm committed:
Further split ensureValidSalesData into two methods

- ensureValidQuantitySold
- ensureValidUnitPrice

Also, update method names to be more descriptive.

xenthm committed:
Parse sales data read from data file

xenthm committed:
Parse manga name and deadline read from data file

averageandyyy committed:
Merge pull request #181 from xenthm/fix-manual-data

Fix manual data

averageandyyy committed:
Merge github.com:AY2425S1-CS2113-T10-3/tp into branch-RefactorUI

averageandyyy committed:
Fix PPP naming

averageandyyy committed:
refactor printing methods in storage to static

xenthm committed:
Check for duplicate manga while deserializing

- Add overloaded ensureNoDuplicateManga method in CommandValidity to be
  used in deserialization. The mangaList needs to be passed explicitly
  as the associated author object is not complete yet.
- Add reference to MangaList in MangaDeserializer and refactor code
  elsewhere
- Change mentions of "corrupt" to "invalid"
- Update JUnit tests of the classes affected

xenthm committed:
Check for duplicate authors while deserializing

    - Remove unused imports
    - Add reference to AuthorList in AuthorDeserializer and refactor code
      elsewhere
    - Change mentions of "corrupt" to "invalid"
    - Update JUnit tests of the classes affected

xenthm committed:
Merge pull request #177 from averageandyyy/branch-RefactorUI

Branch refactor UI

xenthm committed:
Merge remote-tracking branch 'upstream/master' into branch-validity-checks

Add false asserts at parts that need a deeper look.

xenthm committed:
Fix discrepencies in print messages in Ui

Also removed assert false statements.

xenthm committed:
Fix checkstyle and JUnit tests

Commit:
Specified that deadline is overwritten based on comment by @sarahchow03

sarahchow03 committed:
Merge pull request #170 from iaso1774/ped-bug-fixes

Fix issues brought up with Schedule in the PED

xenthm committed:
Rename CommandValidity to CommandValidator

xenthm committed:
Merge pull request #182 from xenthm/branch-validity-checks

Branch validity checks

xenthm committed:
Update UG with duplicate keys in data file limitation

xenthm committed:
Reformat labels in UiClass.puml

xenthm committed:
Update UML diagrams to reflect logger change

Change COMMAND_LOGGER to a public static method.

xenthm committed:
Update code snippets and outputs in DG

xenthm committed:
Update DG

Various fixes:
- Typos
- Formatting

xenthm committed:
Update DG

with:
- CommandValidator reference in the Storage section
- base program location for jar files in the Storage section
- Deadline maximum length (20)

averageandyyy committed:
Update command images with CommandValidator

xenthm committed:
Update UML diagrams

- Moved all diagrams so that they are under their own directory with
  their corresponding puml file
- Regenerated all diagram png files

averageandyyy committed:
Update PPP

averageandyyy committed:
Merge pull request #183 from xenthm/fix-doc

Fix documentation

averageandyyy committed:
Merge github.com:AY2425S1-CS2113-T10-3/tp into branch-FinaliseDG

averageandyyy committed:
Update images with CommandValidator

xenthm committed:
Merge remote-tracking branch 'upstream/master' into branch-ppp

xenthm committed:
Update DG

- Fix "RegEx" typos
- Update table of contents

xenthm committed:
Update AboutUs.md

- Delete template PPP
- Add Ian's file

xenthm committed:
Fix minor typos in DG

xenthm committed:
Update link to UserGuide.md in DG

averageandyyy committed:
Update images

xenthm committed:
Update names in AboutUs.md

xenthm committed:
Update PPP

Cut lines and reformat.

xenthm committed:
Merge pull request #184 from averageandyyy/branch-FinaliseDG

Branch finalise dg

xenthm committed:
Revert to list in PPP

sarahchow03 committed:
Merge pull request #119 from xenthm/branch-ppp

Donovan's PPP

xenthm committed:
Fix link in AboutUs.md

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master'

xenthm committed:
Insert horizontal divider between consecutive diagrams

sarahchow03 committed:
Remove point from note

xenthm committed:
Update Quick Start in UG

Give instructions on how to run the jar file.

xenthm committed:
Fix sales parsing and data reading

- Adding sales data with command did not show the numbers that were just
  added as a message. This was fixed in AddSalesCommand.java
- In SaleDeserializer.java, perform checks on the saleJsonObject before
  proceeding

xenthm committed:
Merge pull request #187 from sarahchow03/branch-fixPPP

Fix Sarah's PPP

sarahchow03 committed:
Merge pull request #185 from xenthm/branch-ppp

Final fixes

sarahchow03 committed:
Merge remote-tracking branch 'upstream/master'

xenthm committed:
Remove redundant spaces in code blocks in UG

sarahchow03 committed:
Merge pull request #188 from xenthm/branch-ppp

Remove redundant spaces in code blocks in UG

xenthm committed:
Try another horizontal line in DG

The previous one did not work.

sarahchow03 committed:
Merge pull request #189 from xenthm/branch-ppp

Try another horizontal line in DG

sarahchow03 committed:
Fix images with cut-off activation bars

xenthm committed:
Merge pull request #190 from sarahchow03/branch-imgFix

Fixes to images in DG

xenthm committed:
Delete README.md

GitHub should look at the docs directory to find the updated README.md

sarahchow03 committed:
Merge pull request #191 from AY2425S1-CS2113-T10-3/xenthm-patch-1

Delete README.md
